spec/ics-3-connection-semantics/README.md

```diff
@@ -0,0 +1,245 @@
+- No chain can masquerade as one of the handshaking chains (formalize...)
+
+#### Post-Establishment
+
+- Connection provides verification of relayed packets
+- No packets other than committed by consensus of *A* / *B* on connection *C* can be relayed
+
+### Technical Specification
+
+(detailed technical specification: syntax, semantics, sub-protocols, algorithms, data structures, etc)
+
+#### Definitions
+
+- Chain `A` is the source blockchain from which the IBC packet is sent
+- Chain `B` is the destination blockchain on which the IBC packet is received
+- `H_h` is the signed header of chain `A` at height `h` 
+- `C_h` is a subset of the consensus ruleset of chain `A` at height `h` 
+- `V_kh` is the value stored on chain `A` under key `k` at height `h` 
+- `P` is the unbonding period of chain `P`, in units of time
+- `dt(a, b)` is the time difference between events `a` and `b` 
+
+Note that of all these, only `H_h` defines a signature and is thus attributable.
+
+#### Requirements
+
+(reference ICS 2: consensus primitives)
+
+To facilitate an IBC connection, the two blockchains must provide the following proofs:
+
+1. Given a trusted `H_h` and `C_h` and an attributable update message `U_h`,  
+   it is possible to prove `H_h'` where `C_h' == C_h` and `dt(now, H_h) < P`
+2. Given a trusted `H_h` and `C_h` and an attributable change message `X_h`,  
+   it is possible to prove `H_h'` where `C_h' /= C_h` and `dt(now, H_h) < P`
+3. Given a trusted `H_h` and a Merkle proof `M_kvh` it is possible to prove `V_kh`
+
+It is possible to make use of the structure of BFT consensus to construct extremely lightweight and provable messages `U_h'` and `X_h'`. The implementation of these requirements with Tendermint consensus is defined in [Appendix E](appendices.md#appendix-e-tendermint-header-proofs). Another algorithm able to provide equally strong guarantees (such as Casper) is also compatible with IBC but must define its own set of update and change messages.
+
+The Merkle proof `M_kvh` is a well-defined concept in the blockchain space, and provides a compact proof that the key value pair `(k, v)` is consistent with a Merkle root stored in `H_h`. Handling the case where `k` is not in the store requires a separate proof of non-existence, which is not supported by all Merkle stores. Thus, we define the proof only as a proof of existence. There is no valid proof for missing keys, and we design the algorithm to work without it. 
+
+Blockchains supporting IBC must implement Merkle proof verification:
+
+`valid(H_h, M_kvh) ⇒ true | false`
+
+#### Lifecycle
+
+##### Opening a connection
+
+All proofs require an initial `H_h` and `C_h` for some `h`, where `dt(now, H_h) < P`.
+
+Establishing a bidirectional initial root-of-trust between the two blockchains (`A` to `B` and `B` to `A`) — `H_ah` and `C_ah` stored on chain `B`, and `H_bh` and `C_bh` stored on chain `A` — is necessary before any IBC packets can be sent.
```

Comment:
We should probably define this notation, ie. `H_ah` means `H_h` for chain `A` ...

Reply:
The handshake protocol has now been thoroughly defined, including datagram types.

There are a lot of additional properties it requires that the old spec never enumerated (primarily MitM-prevention) - these are now discussed.